core/modules/layout/layout.module

```diff
@@ -1684,9 +1698,6 @@ function layout_set_layout_tempstore($item, $type = 'layout') {
     return;
   }
 
-  // Layout contexts should not be stored in the tempstore, they're generated on
-  // the fly when needed after loading.
-  $item->resetContexts();
```

Comment:
It seems like this could stay here based on the new content of `resetContexts()`? Though perhaps it's not necessary with the new approach.